ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,214 @@
+import '@nuxeo/nuxeo-elements/nuxeo-element.js';
+import '../../core/nuxeo-page-provider.js';
+import '../search/nuxeo-search-form-layout.js';
+import '../search/nuxeo-search-results-layout.js';
+import '../widgets/nuxeo-dialog.js';
+import '@polymer/paper-button/paper-button.js';
+
+{
+  /**
+   * `nuxeo-document-picker`
+   * @group Nuxeo UI
+   * @element nuxeo-document-picker
+   */
+  class DocumentPicker extends mixinBehaviors([I18nBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style include="nuxeo-styles">
+          nuxeo-dialog {
+            display: flex;
+            flex-direction: column;
+            justify-content: space-between;
+            height: 100%;
+            width: 100%;
+            /* TODO check if we can use something a bit bigger here */
+            max-height: 84%;
+            max-width: 768px;
+            padding: 8px;
+            overflow: auto;
+          }
+          #topContainer {
+            display: flex;
+            justify-content: flex-end;
+          }
+          iron-icon {
+            opacity: 0.6;
+          }
+          iron-icon:hover {
+            cursor: pointer;
+            opacity: 1;
+          }
+          #mainContainer,
+          #topContainer,
+          nuxeo-results-view {
+            margin: 0;
+            padding: 0;
+          }
+          nuxeo-results-view {
+            --nuxeo-results-view-height: calc(100vh - 520px - var(--nuxeo-app-top));
+          }
+        </style>
+        <nuxeo-dialog id="dialog" modal>
+          <div id="mainContainer">
+            <div id="topContainer">
+              <iron-icon icon="nuxeo:cross" on-tap="close"></iron-icon>
+            </div>
+            <nuxeo-results-view
+              id="resultsView"
+              provider="[[provider]]"
+              page-size="[[pageSize]]"
+              params="[[params]]"
+              quick-filters="{{quickFilters}}"
+              schemas="[[schemas]]"
+              enrichers="[[enrichers]]"
+              search-name="[[searchName]]"
+              aggregations="{{aggregations}}"
+              visible
+              show-filters
+              opened
+              on-navigate="_onNavigate"
+              on-results-changed="_onResultsChanged"
+            ></nuxeo-results-view>
+          </div>
+          <div class="buttons">
+            <!-- TODO check if we can have these stuck to the bottom (absolute positioning? check creation wizard) -->
+            <paper-button noink class="secondary" dialog-dismiss>Cancel</paper-button>
+            <paper-button noink class="primary" on-tap="_onInsert" id="insertButton">Insert</paper-button>
+          </div>
+        </nuxeo-dialog>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-picker';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * The aggregations returned by `provider`.
+         */
+        aggregations: Object,
```

Comment:
this should be private or, if public, set to notify. Same thing for other stuff bound to the page provider.. probably better made "private" for now or set to notify so it fires change events

Reply:
The ones not being used by `<nuxeo-html-editor>` were made private.